tests/multidevice/multidevice.cpp

```diff
@@ -224,4 +224,155 @@ TEST_F(multidevice_ft, ensure_color_camera_is_enabled)
     // Make sure we found both devices.
     ASSERT_EQ(master_device_found, true);
     ASSERT_EQ(subordinate_device_found, true);
-}
\ No newline at end of file
+}
+
+typedef struct _parallel_start_data_t
+{
+    k4a_device_t device;
+    k4a_device_configuration_t *config;
+    bool started;
+    LOCK_HANDLE lock;
+} parallel_start_data_t;
+
+static int parallel_start_thread(void *param)
+{
+    parallel_start_data_t *data = (parallel_start_data_t *)param;
+    k4a_result_t result = K4A_RESULT_SUCCEEDED;
+
+    Lock(data->lock);
```

Comment:
Because you are using this for synchronization, one thread will always be slightly ahead of the other by however long the acquire and unlock of lock is. What this close enough to cause the old implementation to fail?

Reply:
Yes it was, I was able repro it very easily.